web/src/main/webapp/xslt/services/sitemap/sitemap.xsl

```diff
@@ -77,11 +77,10 @@
               <xsl:if test="string($format)"><xsl:value-of select="$format"/>/
               </xsl:if>
             </xsl:variable>
-            <loc><xsl:value-of select="/root/gui/env/server/protocol"/>://<xsl:value-of
-              select="/root/gui/env/server/host"/>:<xsl:value-of
-              select="/root/gui/env/server/port"/><xsl:value-of select="/root/gui/url"/>/sitemap/<xsl:value-of
-              select="$formatParam"/><xsl:value-of select="$pStart"/>/<xsl:value-of
-              select="/root/gui/language"/>
+            <loc>
+              <xsl:value-of select="$siteURL"/><xsl:value-of select="/root/gui/url"/>/sitemap/<xsl:value-of
```

Comment:
I have tested this change and the sitemap I get has relative urls:

```
<loc>/geonetwork/sitemap/1/eng</loc>
```

`siteURL` is defined in `base-variables.xsl` like https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/xslt/common/base-variables.xsl#L100-L101

```
<!-- URL for services - may not be defined FIXME or use fullURL instead -->
<xsl:variable name="siteURL" select="/root/gui/siteURL"/>
```

Not very confident, seem Jeeves doesn't fill it, but depends on each service to fill it, please check if using `XslUtil.getSiteUrl` in the xsl solves this.